### PR TITLE
AB#106560_improve_activity_log_navigation

### PR DESCRIPTION
**Pull request by TaiKamilla**
# Description

This PR filters the activity logs on application or in users, by their corresponding ID's.
This also works for the XLSX file

## Useful links

- [Please insert link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/106560)
- [Please insert link to front-end branch if any](https://github.com/ReliefApplications/ems-frontend/pull/2715)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Improvement (refactor or addition to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [ ] Test A
- [ ] Test B

## Screenshots

https://github.com/user-attachments/assets/e5256fcc-7182-4df9-af79-521bd6a8dfb5

# Checklist:

( * == Mandatory ) 

- [x] * I have set myself as assignee of the pull request
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have put the ticket for review, adding the oort-backend team to the list of reviewers
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

File: src/models/activityLog.model.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import mongoose, { Schema, Document } from 'mongoose';
 export interface ActivityLog extends Document {
   kind: 'ActivityLog';
   userId: mongoose.Types.ObjectId;
+  applicationId: mongoose.Types.ObjectId;
```
Comment (AntoineRelief) [inline src/models/activityLog.model.ts:8]: @TaiKamilla 

I believe the logic is working, but please put the applicationId under metadata, so same level than the url for example

Comment (TaiKamilla) [reply src/models/activityLog.model.ts:8]: Do you want me to remove it from here, or just add it to the metadata too?

Comment (AntoineRelief) [reply src/models/activityLog.model.ts:8]: please remove it

File: src/schema/query/activityLogs.query.ts [hunk 2]
```diff
@@ -53,22 +53,36 @@ type ActivityLogsArgs = {
...
     afterCursor: { type: GraphQLID },
     filter: { type: GraphQLJSON },
+    userId: { type: GraphQLString },
+    applicationId: { type: GraphQLString },
   },
-  async resolve(parent, args: ActivityLogsArgs) {
-    // Make sure that the page size is not too important
+  async resolve(
+    parent,
+    args: ActivityLogsArgs & { userId?: string; applicationId?: string }
```
Comment (AntoineRelief) [inline src/schema/query/activityLogs.query.ts:69]: @TaiKamilla 
the ActivityLogsArgs type is supposed to be extended in order to work with the userId & applicationId
Adding them like how you did is incorrect

**Pull request merged**